### PR TITLE
JDBC

### PR DESCRIPTION
**Pull request by Saranyar04**

### CODE REVIEW

File: src/main/java/org/example/models/hospital/Appointment.java [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
-import org.example.models.persons.Patients;
-import org.example.models.persons.Physicians;
+import org.example.models.persons.Nurse;
+import org.example.models.persons.Patient;
+import org.example.models.persons.Physician;
 
-public class Appointments {
+public class Appointment {
 
     private int appointmantID;
```
Comment (ankochergin) [inline src/main/java/org/example/models/hospital/Appointment.java]:
```suggestion
    private int appointmentID;
```

File: src/main/java/org/example/models/hospital/Appointment.java [hunk 2]
```diff
@@ -1,24 +1,24 @@
...
-    private Patients patients;
-    private Physicians physicians;
-    private Nurses nurses;
-    private ExaminationRooms examinationRooms;
-    private Procedures procedures;
+    private Patient patients;
+    private Physician physicians;
+    private Nurse nurses;
+    private ExaminationRoom examinationRooms;
+    private Procedure procedures;
```
Comment (ankochergin) [inline src/main/java/org/example/models/hospital/Appointment.java]: rename variables in classes and methods too

File: src/main/java/org/example/interfaces/IMedicationsDAO.java [hunk 3]
```diff
@@ -1,10 +1,10 @@
...
-import org.example.models.hospital.Medications;
+import org.example.models.hospital.Medication;
 
 import java.util.List;
 
-public interface IMedicationsDAO extends IBaseDAO<Medications>{
+public interface IMedicationsDAO extends IBaseDAO<Medication>{
 
-    List<Medications> getMedicationsByManufacturer(int manufacturerID);
+    List<Medication> getMedicationsByManufacturer(int manufacturerID);
```
Comment (ankochergin) [inline src/main/java/org/example/interfaces/IMedicationsDAO.java]:
```suggestion
    List<Medication> getMedicationsByManufacturerID(int manufacturerID);
```

File: src/main/java/org/example/interfaces/IExaminationRoomsDAO.java [hunk 4]
```diff
@@ -1,10 +1,10 @@
...
-import org.example.models.hospital.ExaminationRooms;
+import org.example.models.hospital.ExaminationRoom;
 
 import java.util.List;
 
-public interface IExaminationRoomsDAO extends IBaseDAO<ExaminationRooms> {
+public interface IExaminationRoomsDAO extends IBaseDAO<ExaminationRoom> {
 
-    List<ExaminationRooms> getRoomsbyType(String roomType);
+    List<ExaminationRoom> getRoomsbyType(String roomType);
```
Comment (ankochergin) [inline src/main/java/org/example/interfaces/IExaminationRoomsDAO.java]:
```suggestion
    List<ExaminationRoom> getRoomsByType(String roomType);
```

File: src/main/java/org/example/interfaces/IDepartmentsDAO.java [hunk 5]
```diff
@@ -1,10 +1,10 @@
...
-import org.example.models.hospital.Departments;
+import org.example.models.hospital.Department;
 
 import java.util.List;
 
-public interface IDepartmentsDAO extends IBaseDAO<Departments> {
+public interface IDepartmentsDAO extends IBaseDAO<Department> {
 
-    public List<Departments> getDepartmentOrderByName();
+    public List<Department> getDepartmentOrderByName();
```
Comment (ankochergin) [inline src/main/java/org/example/interfaces/IDepartmentsDAO.java]:
```suggestion
    public List<Department> getDepartmentsSortedByName();
```

File: src/main/java/org/example/interfaces/IPatientsDAO.java [hunk 6]
```diff
@@ -1,10 +1,10 @@
...
-import org.example.models.persons.Patients;
+import org.example.models.persons.Patient;
 
 import java.util.List;
 
-public interface IPatientsDAO extends IBaseDAO<Patients> {
+public interface IPatientsDAO extends IBaseDAO<Patient> {
 
-    List<Patients> getPatientbyPysician(int patientID);
+    List<Patient> getPatientbyPysician(int patientID);
```
Comment (ankochergin) [inline src/main/java/org/example/interfaces/IPatientsDAO.java]:
```suggestion
    List<Patient> getPatientByPysician(int physicianID);
```

File: src/main/java/org/example/interfaces/IPositionsDAO.java [hunk 7]
```diff
@@ -1,10 +1,10 @@
...
-import org.example.models.persons.Positions;
+import org.example.models.persons.Position;
 
 import java.util.List;
 
-public interface IPositionsDAO extends IBaseDAO<Positions>{
+public interface IPositionsDAO extends IBaseDAO<Position>{
 
-    List<Positions> getTopThree();
+    List<Position> getTopThree();
```
Comment (ankochergin) [inline src/main/java/org/example/interfaces/IPositionsDAO.java]:
```suggestion
    List<Position> getTopPositions(int limit);
```

File: src/main/java/org/example/models/hospital/Department.java [hunk 8]
```diff
@@ -1,16 +1,14 @@
 package org.example.models.hospital;
 
-import org.example.models.persons.Physicians;
-
-public class Departments {
+public class Department {
 
     private int departmentID;
     private String name;
```
Comment (ankochergin) [inline src/main/java/org/example/models/hospital/Department.java:12]: you can add List<Nurses> and add nurses using DepartmentHaSNurses DAO

File: src/main/java/org/example/models/hospital/Department.java [hunk 9]
```diff
@@ -1,16 +1,31 @@
...
 
+    public List<Nurse> getNursesList() {
+        return nursesList;
+    }
+
+    public void setNursesList(List<Nurse> nursesList) {
+        this.nursesList = nursesList;
+    }
+
+    private List<Nurse> nursesList;
```
Comment (ankochergin) [inline src/main/java/org/example/models/hospital/Department.java:11]: put variables near all other variables

File: src/main/java/org/example/models/hospital/Department.java [hunk 10]
```diff
@@ -1,16 +1,31 @@
...
     private String name;
 
+    public List<Nurse> getNursesList() {
+        return nursesList;
+    }
+
+    public void setNursesList(List<Nurse> nursesList) {
+        this.nursesList = nursesList;
+    }
+
```
Comment (ankochergin) [inline src/main/java/org/example/models/hospital/Department.java]: put to the bottom of the class

File: src/main/java/org/example/interfaces/IDepartmentsDAO.java [hunk 11]
```diff
@@ -6,5 +6,5 @@
 
 public interface IDepartmentsDAO extends IBaseDAO<Department> {
```
Comment (ankochergin) [inline src/main/java/org/example/interfaces/IDepartmentsDAO.java]: rename to IDepartmentDAO, fix it for all classes

File: src/main/java/org/example/jdbc/dao/AppointmentsDAO.java [hunk 12]
```diff
@@ -25,7 +25,7 @@ public class AppointmentsDAO implements IAppointmentDAO {
     private final ProceduresDAO proceduresDAO = new ProceduresDAO();
```
Comment (ankochergin) [inline src/main/java/org/example/jdbc/dao/AppointmentsDAO.java]: rename DAO classes to just `AppointmentDAO` for consistence

File: src/main/java/org/example/interfaces/IDepartmentHasNurseDAO.java [hunk 13]
```diff
@@ -1,10 +1,13 @@
...
 import org.example.models.hospital.DepartmentsHasNurse;
+import org.example.models.persons.Nurse;
 
 import java.util.List;
 
 public interface IDepartmentHasNurseDAO extends IBaseDAO<DepartmentsHasNurse> {
 
-    List<DepartmentsHasNurse> getDepartmentByNurse(int nurseID);
+    List<DepartmentsHasNurse> getDepartmentByNurseID(int nurseID);
+    List<Nurse> addNursesToDepartment(Department department);
```
Comment (ankochergin) [inline src/main/java/org/example/interfaces/IDepartmentHasNurseDAO.java]: here I think you should have method List<Nurse> `getNursesByDepartmentID(int departmentID);` or `List<Nurse> getNursesByDepartment(Department department);`

File: src/main/java/org/example/jdbc/dao/DepartmentsDAO.java [hunk 14]
```diff
@@ -139,7 +139,7 @@ public List<Department> getAll() {
     }
 
     @Override
-    public List<Department> getDepartmentOrderByName() {
+    public List<Department> getDepartmentSortedByName() {
         Connection connection = connectionPool.getConnection();
```
Comment (ankochergin) [inline src/main/java/org/example/jdbc/dao/DepartmentsDAO.java:145]: check it after comment in the IDepartmentHasNurseDAO interface: 
in your getAll and getByID methods you should assign nurses to your class. reuse getNursesByDepartmentID from DepartmentHasNurseDAO

Comment (Saranyar04) [reply src/main/java/org/example/jdbc/dao/DepartmentsDAO.java:145]: when I tried to reuse getNurseByDepartmentID method in department class it leads to recursive call and stack overflow error.
I tried fixing it by declaring as static but not sure if it is the right way. Please suggest me on what can be done

**Pull request merged**